dluhc-component-library/src/components/siteSelectionForm/components/MultiSelect.tsx

```diff
@@ -1,34 +1,41 @@
 interface MultiSelectProps {
   options: ReadonlyArray<string>;
-  values?: Record<string, boolean>;
-  onChange: (values: Record<string, boolean>) => void;
+  values?: Array<string>;
+  onChange: (values: Array<string>) => void;
 }
 
-const MultiSelect = ({ options, values = {}, onChange }: MultiSelectProps) => {
+const MultiSelect = ({ options, values = [], onChange }: MultiSelectProps) => {
   if (!options.length) {
     return null;
   }
 
-  const handleChange = (key: string, value: boolean) => {
-    onChange({
-      ...values,
-      [key]: value,
-    });
+  const handleChange = (value: string, isChecked: boolean) => {
+    let updatedValues;
```

Comment:
`Non-blocker`: this could also be done without this variable. We could do `onChange([...values, value])` and `onChange(values.filter((existingValue) => existingValue !== value))`